src/core/annotation.js

```diff
@@ -922,7 +922,7 @@ class AnnotationBorderStyle {
       this.width = 0; // This is consistent with the behaviour in Adobe Reader.
       return;
     }
-    if (Number.isInteger(width)) {
+    if (Number.isInteger(Math.round(width))) {
```

Comment:
I don't believe that this is entirely correct, since neither https://www.adobe.com/content/dam/acom/en/devnet/pdf/pdfs/PDF32000_2008.pdf#page=392 nor https://www.adobe.com/content/dam/acom/en/devnet/pdf/pdfs/PDF32000_2008.pdf#G11.2096210 explicitly say that the border width has to be an *integer*.
More likely the existing PDF.js code is actually wrong here, and we should accept any *number* instead; i.e. `typeof width === "number"` here.

Reply:
Thank you for the feedback. I'll write a better commit message according to your advice.

It’s exactly as you said,border width can be float as well so I'll change the code to `typeof width === "number"` and add an `eq` test.

Reply:
> More likely the existing PDF.js code is actually wrong here

You're right, and this PR is indeed in the right direction. In the documents we had during the implementation and that we saw so far this has always been an integer, but the specification states "number", which we falsely assumed to be "integer". I just had another look and found the following note (note 2) in section 7.3.3 of the specification at https://www.adobe.com/content/dam/acom/en/devnet/pdf/pdfs/PDF32000_2008.pdf#page=22&zoom=auto,-215,391:

> Throughout this standard, the term number refers to an object whose type may be either integer or real.

In short, this PR is fixing the issue in the right place compared to other attempts, and I completely agree with the suggestions from @Snuffleupagus.

Reply:
Thank you for the review.
>Aside from the eq test, it would also be good to adjust the already existing unit tests to check that float inputs are also handled correctly.

Considering your advice I have checked `test/pdf` and there is [annotation-border-styles.pdf](https://github.com/mozilla/pdf.js/files/7769542/annotation-border-styles.pdf) (attached) in the directory it is pdf with annotation borders and there are lot of blank space.

I thought it would be good to put an annotation with a non-integer border width in it.But I think it would be good to add pdf which name is `annotation-border-width.pdf ` because it fully describes what the pdf stands for so I would add an pdf.

Reply:
> Considering your advice I have checked `test/pdf` and there is [annotation-border-styles.pdf](https://github.com/mozilla/pdf.js/files/7769542/annotation-border-styles.pdf) (attached) in the directory.It is pdf with annotation borders and there are lot of blank space in it.

Please do *not* update an existing test document, since there's a very real possibility that you could either break the file or possibly no longer test the originally intended thing.
Instead, please add a new *linked* test-case using the PDF document from issue #14203 and a `test_manifest.json` entry looking something like this:
```
{  "id": "issue14203",
   "file": "pdfs/issue14203.pdf",
   "md5": "26c1c1e7341bc23fb28d805fa516dbbe",
   "link": true,
   "rounds": 1,
   "firstPage": 11,
   "lastPage": 11,
   "type": "eq"
},
```